libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -575,13 +575,14 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 					"Only internal function calls implemented for libraries"
 				);
 
+	std::map<ASTPointer<Expression const>, string> expressionCache;
```

Comment:
I think a little code duplication for addmod and mulmod would not hurt, but this variable here makes code generation rather intransparent.

Reply:
you're right. especially if it is only used for addmod and mulmod.